include/swift/Runtime/HeapObject.h

```diff
@@ -84,8 +84,12 @@ struct TwoWordPair {
   // in registers, so cram the result into an unsigned long long.
   // Use an enum class with implicit conversions so we don't dirty C callers
   // too much.
-#if __arm__ || __i386__
+#if __arm__ || __i386__ || defined(__CYGWIN__)
+#if defined(__CYGWIN__)
+  using Return = unsigned __int128;
```

Comment:
Can you use `enum class Return : unsigned __int128`, or is that not supported? It'd be good to avoid implicit conversions from raw integer types.